torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,800 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# FLAIR dataset is realeasd under open license 2.0
+# ..... https://www.etalab.gouv.fr/wp-content/uploads/2018/11/open-licence.pdf
+# ..... https://ignf.github.io/FLAIR/#FLAIR2
+
+"""FLAIR2 dataset."""
+
+import glob
+import json
+import os
+from collections.abc import Callable, Collection, Sequence
+from typing import Any, ClassVar, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.colors import ListedColormap
+from matplotlib.figure import Figure
+from matplotlib.patches import Patch, Rectangle
+from torch import Tensor
+
+from .errors import DatasetNotFoundError, RGBBandsMissingError
+from .geo import NonGeoDataset
+from .utils import Path, check_integrity, download_url, extract_archive
+
+
+class FLAIR2(NonGeoDataset):
+    """FLAIR #2 (The French Land cover from Aerospace ImageRy) dataset.
+
+    The `FLAIR #2 <https://github.com/IGNF/FLAIR-2>` dataset is an extensive dataset from the French National Institute
+    of Geographical and Forest Information (IGN) that provides a unique and rich resource for large-scale geospatial analysis.
+    The  dataset is sampled countrywide and is composed of over 20 billion annotated pixels of very high resolution aerial
+    imagery at 0.2 m spatial resolution, acquired over three years and different months (spatio-temporal domains).
+
+    The FLAIR2 dataset is a dataset for semantic segmentation of aerial images. It contains aerial images, sentinel-2 images and masks for 13 classes.
+    The dataset is split into a training and test set.
+
+    Dataset features:
+
+    * over 20 billion annotated pixels
+    * aerial imagery
+        * 5x512x512
+        * 0.2m spatial resolution
+        * 5 channels (RGB-NIR-Elevation)
+    * Sentinel-2 imagery
+        * 10-20m spatial resolution
+        * 10 spectral bands
+        * snow/cloud masks (with 0-100 probability)
+        * multiple time steps (T)
+        * Tx10xWxH, T, W, H are variable
+    * label (masks)
+        * 512x512
+        * 13 classes
+
+    Dataset classes:
+
+    0: "building",
+    1: "pervious surface",
+    2: "impervious surface",
+    3: "bare soil",
+    4: "water",
+    5: "coniferous",
+    6: "deciduous",
+    7: "brushwood",
+    8: "vineyard",
+    9: "herbaceous vegetation",
+    10: "agricultural land",
+    11: "plowed land",
+    12: "other"
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.48550/arXiv.2310.13336
```

Comment:
Should probably cite the FLAIR #2 paper (https://arxiv.org/abs/2305.14467) instead of the FLAIR paper